pkgs/by-name/ge/gettext/package.nix

```diff
@@ -0,0 +1,125 @@
+{ lib
+, bash
+, darwin
+, fetchurl
+, libiconv
+, stdenv
+, xz
+
+, enableCoreServices ? stdenv.isDarwin
+# HACK, see #10874 (and 14664)
+, enableLibiconv ? (!stdenv.isLinux && !stdenv.hostPlatform.isCygwin)
+}:
+
+# Note: this package is used for bootstrapping fetchurl, and thus cannot use
+# fetchpatch! All mutable patches (retrieved via by GitHub, cgit or similar
+# means) that perhaps are needed here should be included directly in Nixpkgs as
+# regular files inside this same directory.
+
+stdenv.mkDerivation (finalAttrs: ({
+  pname = "gettext";
+  version = "0.22.5";
+
+  src = fetchurl {
+    url = "mirror://gnu/gettext/gettext-${finalAttrs.version}.tar.gz";
+    hash = "sha256-7BcFselpuDqfBzFE7IBhUduIEn9eQP5alMtsj6SJlqA=";
+  };
+
+  patches = [
+    # Parameterize some tools to their absolute paths in gettext.sh.in
+    ./001-absolute-paths.diff
+    # fix reproducibile output, in particular in the grub2 build
+    # https://savannah.gnu.org/bugs/index.php?59658
+    ./002-msginit-do-not-use-pot-creation-date.patch
+  ]
+  ++ lib.optionals enableCoreServices [
+    # prevent infinite recursion for the darwin stdenv
+    ./003-revert-avoid-crash-on-macos-14.patch
+  ];
```

Comment:
```suggestion
  ];
  
  propagatedBuildInputs = lib.optionals enableCoreServices [
    # prevent infinite recursion for the darwin stdenv
    ./003-revert-avoid-crash-on-macos-14.patch
  ];
```

Reply:
When `autoreconfPhase` is run with the new gettext, `m4/intlmacosx.m4` is updated, adding CoreServices as a dependency of the original package.

Reply:
After rebuilding the stdenv with this change, there is one downside. It requires adding CoreServices and its propagated inputs the Darwin stdenv’s `allowedRequisites`, which differ depending on the SDK.

Instead of using `propagatedBuildInputs`, is it possible to add a `dev` output to gettext, keep CoreServices as a regular `buildInput`, and add a conditional `postInstall` to propagate it from the dev output?

I used the following (with CoreServices as a regular build input) to successfully rebuild the Darwin stdenv. libpsl is an example of a package that needs CoreServices propagated to it. It also built successfully.

I also split the hook into two hooks. The one copied into `$out` sets up `GETTEXTDATADIRS` while the other ensures that `libintl` is linked.

```nix
  # Two setup hooks are needed. One is for making sure gettext can find its data files at runtime, and the other
  # is to make sure `libintl` is linked even on non-glibc targets. This is the latter hook. The other is below.
  setupHooks = [
    ../../../build-support/setup-hooks/role.bash
    ./gettext-setup-hook-dev.sh
  ];

  postInstall = ''
    mkdir -p "$out/nix-support"
    cat ${../../../build-support/setup-hooks/role.bash} ${./gettext-setup-hook.sh} > "$out/nix-support/setup-hook"
  ''
  + lib.optionalString enableCoreServices ''
    mkdir -p "''${!outputDev}/nix-support"
    echo '${darwin.apple_sdk.frameworks.CoreServices}' >> "''${!outputDev}/nix-support/propagated-build-inputs"
  '';
```

Reply:
I will wait your modifications before migrating this, since I don't and I won't have a Darwin machine.

Reply:
Modifications to the existing gettext package? I can open a PR (or PRs, depending) for those.

Reply:
The migration to by-name  will be a little bit harder.

Reply:
Because of the hooks? I cheated with libiconv-darwin (by reexporting them from libiconvReal and using those), but I don’t think that will be possible here.